File: library-net/src/main/java/com/my/library_net/interceptor/HeaderInterceptor.java
package com.my.library_net.interceptor;

import java.io.IOException;

import okhttp3.Interceptor;
import okhttp3.Request;
import okhttp3.Response;

public class HeaderInterceptor implements Interceptor {
    @Override
    public Response intercept(Chain chain) throws IOException {
        Request originalRequest = chain.request();
        Request.Builder builder = originalRequest.newBuilder();
        //设置具体的header内容
        builder.header("timestamp", System.currentTimeMillis() + "");

        Request.Builder requestBuilder =
                builder.method(originalRequest.method(), originalRequest.body());
        Request request = requestBuilder.build();
        return chain.proceed(request);
    }
}

File: library-net/src/main/java/com/my/library_net/interceptor/HeaderInterceptor.kt
package com.my.library_net.interceptor

import okhttp3.Interceptor
import okhttp3.Request
import okhttp3.Response
import java.io.IOException

class HeaderInterceptor constructor() : Interceptor {
    @Throws(IOException::class)
    override fun intercept(chain: Interceptor.Chain): Response {
        val originalRequest: Request = chain.request()
        val builder: Request.Builder = originalRequest.newBuilder()
        //设置具体的header内容
        builder.header("timestamp", System.currentTimeMillis().toString() + "")
        val requestBuilder: Request.Builder = builder.method(originalRequest.method(), originalRequest.body())
        val request: Request = requestBuilder.build()
        return chain.proceed(request)
    }
}